src/Enforcers/ValidationEnforcer.php

```diff
@@ -34,9 +34,21 @@ public function enforce(mixed $object, string $className): void
         $validationErrors = new ValidationListException();
         $metadata = $this->getValidationMetadata($className);
 
+        // If an incoming property is not included on the
+        // object, throw an error for stricter enforcement.
+        $nonExistentProperties = $this->checkForPropertiesNotOnDataObject($metadata, $arr);
+
+        foreach ($nonExistentProperties as $badProperty) {
+            $validationErrors->addError(
+                $this->getErrorMessage($badProperty),
+                $badProperty,
+            );
+        }
+
         //For each property defined on the class.
         foreach ($metadata as $propName => $propMeta) {
-            //If the property is not included on the object skip it.
+            // If the propName from the data-object is not included
+            // on the incoming object, don't attempt to validate it.
             if ($this->propIsUndefined($arr, $propName)) {
                 continue;
```

Comment:
based on the way the previous comment read, we thought this was where we could introduce a "extra property" check but in fact, this is the opposite. It's getting the metadata/attribute info from the data-object and iterating over those properties and checking if they exist in the incoming data.